protocols/raft/src/main/java/io/atomix/protocols/raft/roles/AbstractAppender.java

```diff
@@ -238,6 +238,7 @@ else if (response.term() > raft.getTerm()) {
     else {
       resetMatchIndex(member, response);
       resetNextIndex(member, response);
+      resetSnapshotIndex(member, 0);
```

Comment:
I am wondering if it makes more sense to encode `lastSnapshotIndex` of `member` in the response. `resetSnapshotIndex(member, response.lastSnapshotIndex())`. In case of full data loss this value will be 0. Otherwise it will be the actual snapshot index.

Reply:
So I did a small test to see how resetting `snapshotIndex` to 0 affect the scenario where there is no data loss. The setup has a slow server, which might lose some append requests. In this case there is no restart. Just that one follower(network connection) is slow. In this case, I observed that the leader sends the same snapshot multiple times to this slow follower. It doesn't lead to any inconsistent state. The follower is capable of handling redundant requests. But it can unnecessarily put additional load on the already slow network to this follower. So I would like to see whether the above approach I mentioned works.

Reply:
I thought about that, but I figured this would be a more acceptable change for the maintainers, as it's pretty unobstrusive; I figured we're adding a field to the response which is only used some times, so they might not accept it. I admit that's kind of a weak argument though :sweat_smile: